File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(input) do
    cond do
      silent?(input)   -> "Fine. Be that way!"
      question?(input) -> "Sure."
      shouting?(input) -> "Woah, chill out!"
      :otherwise       -> "Whatever."
    end
  end

  defp shouting?(input), do: input == String.upcase input

  defp question?(input), do: "?" == String.last input

  defp silent?(nil),   do: true
  defp silent?(input), do: "" == String.strip input
end

File: clojure/bob/bob.clj
(ns bob
  (:require [clojure.string :refer [upper-case]]))

(defn- silent? [what]
  (= "" what))

(defn- shout? [what]
  (= what (upper-case what)))

(defn- question? [what]
  (= \? (last what)))

(defn response-for [what]
  (cond
    (silent? what) "Fine, be that way."
    (shout? what) "Woah, chill out!"
    (question? what) "Sure."
    :else "Whatever."))
